### ISSUE TITLE
Forbid nested comprehensions

### ISSUE THREAD

**Issue by orsinium** [wemake-services/wemake-python-styleguide#806]
# Rule request

## Thesis

Python allows the use of nested comprehensions with 2 or more cycles:

```python
[i + j for i in list1 for j in list2]
[val for sublist in matrix for val in sublist]
```

Let's forbid it.

## Reasoning

I think, this construction makes code more difficult to read:

1. The order of cycles isn't intuitive. When you have nested comprehensions (`[[i + j for i in list1] for j in list2]`), you will get cycles order from right to left, but that construction makes it left to right.
1. Comprehensions should be used only for simple and short constructions. This is one-liner to make it easier to write the code. When you have two cycles, that's not the case. The nested comprhension makes code line too complex. 

## Alternative

Rewrite it into nested cycles:

```python
result = []
for i in list1:
  for j in list2:
    result.append(i+j)
```
Or as `itertools.product` cycle:

```python
result = []
for i, j, k in product(list1, list2, list3):
    result.append(i+j+k)
```

**Comment by sobolevn** [comment]
After thinking about this rule for sometime, I made a decision to decline this suggestion. Sorry.

Why?
1. Because we already have `WPS224` that forbids to use more than two `for` loops inside a comprehension
2. Because we use two `for`s a lot in our code base; and it seems fine by me

Anyway, thanks for your time and effort to put this together! I have made a huge contribution to the project with providing new rule ideas and I looking forward to see new ones! 👍
